flink/src/main/java/org/apache/iceberg/flink/connector/sink/AvroSerializer.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink.connector.sink;
+
+import java.io.Serializable;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.IndexedRecord;
+
+/**
+ * Serialize input data type to Avro IndexedRecord
+ */
+@FunctionalInterface
+public interface AvroSerializer<IN> extends Serializable {
```

Comment:
I don't think that the Flink/Iceberg API should use or expose Avro.

Iceberg has its own generic classes that include built-in support for maps with non-string keys, decimals, date/time types, etc. Avro does not easily support those. In addition, Avro is a source of conflicts and is often shaded (including in this PR) so we don't want to use as part of the public API.

I think by default, this should accept Iceberg generic records. We can also add translation to and from Flink's native record format, like we have for Spark's InternalRow.

Reply:
Addressed. Avro IndexedRecord is removed and [Record](https://github.com/apache/incubator-iceberg/blob/master/data/src/main/java/org/apache/iceberg/data/Record.java) is used instead.
Regarding the translation from/to Flink format, will open a new issue to address that.